### PR TITLE
EQL: Protocol support for async mode

### PR DESCRIPTION
**Pull request by imotov**
Adds a protocol support for async mode. The actual implementation
will be added when more details of query handling are clear.

Relates to #49581

### CODE REVIEW

**Comment by imotov** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java [hunk 1]
```diff
@@ -38,10 +38,13 @@
         "   }" +
         "}";
 
-    static String defaultTestIndex = "endgame-*";
+    private String[] defaultTestIndex;
+    private boolean waitForCompletion;
 
     @Before
     public void setup() {
+        defaultTestIndex = generateRandomStringArray(10, 10, false);
```
Comment (aleksmaus) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java]: thanks

**Review by aleksmaus — APPROVED** [review]
LGTM, Thank you!

**Comment by imotov** [comment]
We need to revisit the async approach here so I am going to close this for now. I have extracted the changes to make response less mutable into #50810.

**Pull request closed without merging**